source/lib/permissions-service.js

```diff
@@ -1,19 +1,17 @@
-export async function queryPermission(permission) {
-	const granted = await browser.permissions.contains({permissions: [permission]});
-
-	if (browser.runtime.lastError) {
-		throw new Error(browser.runtime.lastError);
+export function queryPermission(permission) {
+	try {
+		return browser.permissions.contains({permissions: [permission]});
+	} catch (error) {
+		console.log(error);
+		return false;
 	}
-
-	return granted;
 }
 
-export async function requestPermission(permission) {
-	const granted = await browser.permissions.request({permissions: [permission]});
-
-	if (browser.runtime.lastError) {
-		throw new Error(browser.runtime.lastError);
+export function requestPermission(permission) {
+	try {
+		return browser.permissions.request({permissions: [permission]});
+	} catch (error) {
```

Comment:
Why could this fail?

Reply:
It might fail because of https://github.com/sindresorhus/notifier-for-github/pull/180#issuecomment-501558819.

`runtime.lastError` is Chrome specific API, and I'm not seeing Chrome complaining about it, so dropped it.

Reply:
This isn’t resolved, that part can be fixed. 

What I meant is that we in fact do not need to check that property because the Promised API polyfill does it automatically. 

The extension seems full of these checks in places where it doesn’t make sense. It could use a cleanup in a separate PR